src/parsec.rs

```diff
@@ -161,14 +236,35 @@ impl ParsecMap {
 
     #[cfg(feature = "mock_base")]
     pub fn has_unpolled_observations(&self) -> bool {
-        let parsec = if let Some(parsec) = self.map.values().last() {
+        let parsec = if let Some(parsec) = self.get_parsec_values().last() {
             parsec
         } else {
             return false;
         };
 
         parsec.has_unpolled_observations()
     }
+
+    fn get_parsec_mut(&mut self, version: u64) -> Option<&mut Parsec> {
```

Comment:
Might as well be `get_last_parsec_values`\`get_last_parsec_values_mut`

Reply:
It's not the last one though?

Reply:
I put the comment at the wrong place, it was refering to `get_parsec_values` and the nut version

Reply:
The `get_parsec_values` method returns all parsecs though, not the last one

Reply:
Yes, but all the uses follow it with `.last()` so you may as well fold that call within that function and clarify that we only care about the last one.